### ISSUE TITLE
Allow common cross-environment configuration

### ISSUE THREAD

**Issue by janstenpickle** [shantytown/shanty#21]
Currently shanty only supports config per environment, but some plugins will require global configuration which exists across all environments. Propose something like this:

```yaml
---
- all:
  - docker:
    - nic: cage
- local
  - docker
    - nic: coppola
```

**Comment by nathankleyn** [comment]
:+1: 

Perhaps an `all` key rather than `common? I hate the word `common`, it's overused. :-)

**Comment by janstenpickle** [comment]
:+1: Good plan

**Comment by nathankleyn** [comment]
Fixed by #36.
